pkg/cloud/services/eks/nodegroup.go

```diff
@@ -147,9 +148,8 @@ func (s *NodegroupService) remoteAccess(data *ec2.RequestLaunchTemplateData) err
 		)
 	}
 
-	if !pool.RemoteAccess.Public {
-		// TODO create security group for nodes
-		sSGs = pool.RemoteAccess.SourceSecurityGroups
+	if !pool.RemoteAccess.Public && pool.RemoteAccess.SourceSecurityGroups != nil {
```

Comment:
The issue here is that we need to create a _new_ SG for this nodegroup and then allow ingress to _that SG_ from the `SourceSecurityGroups` (my code was totally missing this) @richardcase

Reply:
Unfortunately, using [both launch templates and RemoteAccess is disallowed](https://docs.aws.amazon.com/eks/latest/userguide/launch-templates.html#w626aac16b9c27b7c11)

Reply:
I need some help to start a refactor. I do not have much knowledge about code base and I do not feel confident yet for do it

Reply:
No worries, I'm going to take another look ASAP